### DISCUSSION TITLE
Splitting a large XML message into groups of 5/10

### DISCUSSION THREAD

**Discussion by sharmadeepak69** [nextgenhealthcare/connect#5125] [Q&A]
We are using a database reader to query postgresDB. We get input message in the form of huge XML data as we are aggregating the results. Then, we split this huge message into small batches using split batch using 'level'. But as we get 100k records, it is taking huge amount of time to process those 100k messages.
We thought of breaking the XML aggregated results into batches of 5/10 records each and then develop some logic that can accommodate multithreading, but we are unable to split the XM input into groups of 5. For example -

```
<results>
<result>
<result>
<result>
<result>
<result>
<result>
<result>
<result>
<result>
<result>
</results>

```
into

```
<results>
<result>
<result>
<result>
<result>
<result>
</results>
<results>
<result>
<result>
<result>
<result>
<result>
</results>
```

Can someone please help us in breaking this XML input using batch splitting code via Javascript or XPath Query.

**Comment by siddharthsharma9258** [comment]
why not adjust your db select query and pick only 10 messages per cycle? 
Obviously 100k rows in one cycle will take aweful amount of time from start to end.

**Comment by tonygermano** [comment]
Have you tried not aggregating results? There's not much difference between 1 and 5 records per message when you are looking at 100k total.

Another option is to use a regular javascript reader and skip the xml entirely. It's very likely that if you processed the database results into one large javascript array of objects that the JSON would serialize and deserialize much faster than the xml does.
